spec/lib/fastly_nsq/message_queue/consumer_spec.rb

```diff
@@ -17,49 +14,47 @@ def fake_consumer
     end
   end
 
-  describe 'when using the real queue', fake_queue: false do
-    before(:example) do
-      @fake_consumer = fake_consumer
-      allow(Nsq::Consumer).to receive(:new).and_return(@fake_consumer)
+  describe 'when connector connects to a backend Consumer' do
+    let(:consumer) do
+      MessageQueue::Consumer.new topic: topic, channel: channel, connector: ->(_) { backend }
     end
 
-    it 'forwards #pop to Nsq::Consumer' do
+    it 'forwards #pop' do
+      expect(backend).to receive(:pop)
       consumer.pop
-      expect(@fake_consumer).to have_received(:pop)
     end
 
-    it 'forwards #size to Nsq::Consumer' do
+    it 'forwards #pop_without_blocking' do
+      expect(backend).to receive(:pop_without_blocking)
+      consumer.pop_without_blocking
```

Comment:
No worries - I take the point for sure :)

On Thu, May 12, 2016 at 12:33 PM, Joshua Wehner <notifications@github.com>
wrote:

> In spec/lib/fastly_nsq/message_queue/consumer_spec.rb
> <https://github.com/fastly/fastly_nsq/pull/32#discussion_r63082574>:
>
> >      end
> >
> > -    it 'forwards #size to Nsq::Consumer' do
> > +    it 'forwards #pop_without_blocking' do
> > +      expect(backend).to receive(:pop_without_blocking)
> > +      consumer.pop_without_blocking
>
> No worries, always good to have another TIL thing.
>
> That said ... I dunno about these, I have mixed feelings about
> "specialized" matchers. I think it makes sense when they come from the "app
> domain" (I dunno... "expect to have_connected_to_backend" maybe?) but, for
> me, shoulda_matchers feel like they come from the "rspec domain" (or,
> best case scenario, the "I am writing ruby domain"), which doesn't exactly
> sit right with me. ¯\_(ツ)_/¯
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/fastly/fastly_nsq/pull/32/files/eece8723e93aa9b48166019dfebfea510f185f02#r63082574>
>



-- 
Adarsh Pandit
External Consultant
+1 313 454 1515 (m)
@adarshp <https://twitter.com/adarshp>